packages/discovery-react-components/src/components/DiscoverySearch/__tests__/DiscoverySearch.test.tsx

```diff
@@ -161,14 +161,18 @@ describe('DiscoverySearch', () => {
         fullTree,
         result: { getByTestId, rerender }
       } = setup({ overrideCollectionsResults: { collections: [{ name: 'foo' }] } }, tree);
+
       expect(getByTestId('value').textContent).toEqual('foo');
+
       rerender(
         cloneElement(fullTree, { overrideCollectionsResults: { collections: [{ name: 'bar' }] } })
       );
       expect(getByTestId('value').textContent).toEqual('bar');
+
+      await wait(); // wait for component to finish rendering (prevent "act" warning)
```

Comment:
this is strange - why are we awaiting after expectations?

Reply:
This is basically to avoid React throwing an error saying "Your component did more stuff after it should have finished. Consider wrapping with act()."

So I add this `wait` call to let the test know that there's a little bit more left in the component re-rendering.